### DISCUSSION TITLE
C++ File handling

### DISCUSSION THREAD

**Discussion by HP-hamajis** [community/community#122192] [Programming Help]
### Select Topic Area

General

### Body

what the reason  we use tell g and seek g pointers

**Comment by nafargi** [comment]
In C++ programming, "tellg" and "seekg" are member functions of the std::ifstream class, which is used for input operations on files. Here's what they do:

1. **tellg**: This function returns the current position of the get pointer (read position) in the file stream. It allows you to determine where you are within the file at any given moment. This is particularly useful when you need to later return to that position in the file.

2. **seekg**: This function is used to move the get pointer (read position) to a specific position within the file stream. You can use it to navigate to a specific byte offset within the file or to a position relative to the beginning, current position, or end of the file.

These functions are often used when you need to read data from a file at specific locations or when you need to move around within a file during input operations. For example, you might use them when reading structured data from a file where you need to skip certain sections or when you need to rewind and reread a portion of the file.
